packages/runtime/container-runtime/src/opLifecycle/opDecompressor.ts

```diff
@@ -44,12 +45,18 @@ export class OpDecompressor {
             const asObj = JSON.parse(intoString);
             this.rootMessageContents = asObj;
 
-            return { ...message, contents: this.rootMessageContents[this.processedCount++] };
+            return {
+                message: { ...message, contents: this.rootMessageContents[this.processedCount++] },
+                state: "Accepted",
```

Comment:
I wonder if we can add an extra layer into processing pipeline that would do grouping & ungrouping of op content in a batch that happens here, such that compression code does not know anything about it. To make layers more simple and single-purpose. Not sure if it makes tracking harder (as today we lump together compression and batching, so we do not handle extra states that might happen if we separate them, like no compression but with grouping of content into single op).
If possible, I'd separate into more independent layers, but somehow reduce number of states we need to handle. I.e. maybe we say that both chunking and grouping happens only if compression is enabled.